src/platform/intel/ace/include/ace/lib/mailbox.h

```diff
@@ -80,12 +80,12 @@ static inline uint64_t mailbox_sw_reg_read64(size_t offset)
 
 static inline void mailbox_sw_regs_write(size_t offset, const void *src, size_t bytes)
 {
-	int regs_write_err __unused = memcpy_s((void *)(MAILBOX_SW_REG_BASE + offset),
-					       MAILBOX_SW_REG_SIZE - offset, src, bytes);
+	uint32_t __sparse_cache *ptr_c;
+	volatile uint32_t *ptr;
```

Comment:
volatile inserts a memory barrier when ptr is dereferenced, but in this case I think its compiled out with the call to ```memcpy_s()```.

Reply:
thanks